src/stores/messages/SavedMessagesViewTypesStore.ts

```diff
@@ -0,0 +1,44 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { action, observable } from 'mobx';
+import { SavedMessageViewType } from './SavedMessageViewType';
+
+class SavedMessagesViewTypesStore {
+	@observable
+	public savedViewTypes = new Map<string, SavedMessageViewType>();
+
+	@action
+	public saveViewType = (id: string, vt: SavedMessageViewType) => {
+		if (!this.savedViewTypes.has(id)) {
+			this.savedViewTypes.set(id, vt);
+		}
+	};
+
+	@action
+	public getSavedViewType = (id: string) => {
+		return this.savedViewTypes.get(id);
```

Comment:
and if viewStore wasnt initiated and its undefined we can create an instance of ViewStore, and return it